core/src/main/java/com/nerds/gamejam/ecs/system/RenderSystem.java

```diff
@@ -4,35 +4,31 @@
 import com.artemis.ComponentMapper;
 import com.artemis.annotations.Wire;
 import com.artemis.systems.IteratingSystem;
-import com.badlogic.gdx.Gdx;
 import com.badlogic.gdx.graphics.Color;
-import com.badlogic.gdx.graphics.g2d.Animation;
 import com.badlogic.gdx.graphics.g2d.Batch;
 import com.badlogic.gdx.graphics.g2d.BitmapFont;
-import com.badlogic.gdx.graphics.g2d.Sprite;
 import com.badlogic.gdx.graphics.g2d.SpriteBatch;
 import com.badlogic.gdx.graphics.g2d.TextureRegion;
-import com.nerds.gamejam.ecs.component.AnimationComponent;
-import com.nerds.gamejam.ecs.component.CompositeSpriteComponent;
-import com.nerds.gamejam.ecs.component.FontComponent;
-import com.nerds.gamejam.ecs.component.PositionComponent;
-import com.nerds.gamejam.ecs.component.RenderableComponent;
-import com.nerds.gamejam.ecs.component.SpriteComponent;
+import com.nerds.gamejam.ecs.component.*;
+import com.nerds.gamejam.util.TextureLoader;
 
 @Wire
 public class RenderSystem extends IteratingSystem {
 
-    private ComponentMapper<CompositeSpriteComponent> compositeSpriteMapper;
-    private ComponentMapper<SpriteComponent> spriteMapper;
     private ComponentMapper<PositionComponent> positionMapper;
+    private ComponentMapper<TextureReferenceComponent> textureReferenceMapper;
+    private ComponentMapper<ScaleComponent> scaleMapper;
+    private ComponentMapper<BodyComponent> bodyMapper;
     private ComponentMapper<FontComponent> fontMapper;
-    private ComponentMapper<AnimationComponent> animationMapper;
+    private final TextureLoader textureLoader;
     private CameraSystem cameraSystem;
     private Batch batch;
     private BitmapFont font;
 
-    public RenderSystem() {
-        super(Aspect.all(RenderableComponent.class));
+    public RenderSystem(CameraSystem cameraSystem, TextureLoader textureLoader) {
```

Comment:
Camera system will be injected if you use @Wire on the class
I'd stick with a static use of the TextureLoader tbh